### PR TITLE
EWS: Fixed problem with duplicated site codes in location sync

### PR DESCRIPTION
**Pull request by kkrampa**

### CODE REVIEW

File: custom/ewsghana/api.py [hunk 1]
```diff
@@ -332,11 +335,25 @@ def _set_up_supply_point(self, location, ews_location):
             location.name = supply_point.name
             location.site_code = supply_point.code
             location.location_type = supply_point.type
-            self._create_supply_point_from_location(supply_point, location)
+            try:
+                sql_location = SQLLocation.objects.get(domain=self.domain, site_code=supply_point.code)
```
Comment (kkrampa) [inline custom/ewsghana/api.py]: There are some districts that have the same site code as supply points. I agreed with Mike that they are more important for supply points.

File: custom/ewsghana/api.py [hunk 2]
```diff
@@ -313,6 +313,7 @@ def _set_up_supply_point(self, location, ews_location):
             supply_point_with_stock_data = filter(
                 lambda x: x.last_reported and x.active, ews_location.supply_points
             )
+            location.save()
```
Comment (czue) [inline custom/ewsghana/api.py]: why was this added?

File: custom/ewsghana/api.py (hunk 1 above)
Comment (czue) [inline custom/ewsghana/api.py]: how do you know that you are clearing the code from a district? is it just based on knowing the order that these are called?

File: custom/ewsghana/tests/test_locations_sync.py [hunk 3]
```diff
@@ -109,3 +113,53 @@ def test_facility_with_inactive_and_active_supply_point(self):
...
+        ewsghana_location = self.api_object.location_sync(location)
+        self.assertEqual(ewsghana_location.name, "Test country")
+        self.assertEqual(ewsghana_location.site_code, "testcountry")
+
+        location.name = "edited"
+        location.code = "edited"
+        ewsghana_location = self.api_object.location_sync(location)
+
+        self.assertEqual(ewsghana_location.name, "edited")
+        self.assertEqual(ewsghana_location.site_code, "edited")
```
Comment (czue) [inline custom/ewsghana/tests/test_locations_sync.py]: :+1: tests!

File: custom/ewsghana/api.py (hunk 2 above)
Comment (kkrampa) [inline custom/ewsghana/api.py]: I moved save to _set_up_supply_point because in main location_sync method we don't know which site_code should be used(for facility we are using code from supply point).

File: custom/ewsghana/api.py [hunk 4]
```diff
@@ -332,11 +335,29 @@ def _set_up_supply_point(self, location, ews_location):
             location.name = supply_point.name
             location.site_code = supply_point.code
             location.location_type = supply_point.type
-            self._create_supply_point_from_location(supply_point, location)
+            try:
+                sql_location = SQLLocation.objects.get(domain=self.domain, site_code=supply_point.code)
+                if not sql_location.location_type.administrative:
+                    raise MigrationException(
```
Comment (kkrampa) [inline custom/ewsghana/api.py]: It shouldn't happened but raises exception if there is a conflict.
